### PR TITLE
SAK-44410 Unboundid - add options for setHealthCheckIntervalMillis an…

### PR DESCRIPTION
**Pull request by austin48**
…d setHealthCheck

### CODE REVIEW

File: providers/component/src/webapp/WEB-INF/unboundid-ldap.xml [hunk 1]
```diff
@@ -144,6 +144,30 @@
...
+			Requires healthCheckMappings's invokeForBackgroundChecks to be set to true
+			Defaults to 60000L -->
+		<!-- property name="healthCheckIntervalMillis">
+			<value>60000</value>
+		</property -->
+
+		<!-- Provides mapping to create an LDAP connection pool health check implementation
+			that may be used to check the health of the associated server by verifying that
+			a specified entry can be retrieved in an acceptable period of time
+			Defaults to null (invokes the Unboundid defaut implementation, which may not do any checks) -->
```
Comment (ottenhoff) [inline providers/component/src/webapp/WEB-INF/unboundid-ldap.xml:158]: this looks good. can you be more definitive with this sentence ("may not")? what happens if this is unset?

File: providers/unboundid/src/java/org/sakaiproject/unboundid/UnboundidDirectoryProvider.java [hunk 2]
```diff
@@ -1219,6 +1239,36 @@ public void setRetryFailedOperationsDueToInvalidConnections(boolean retryFailedO
...
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public void setHealthCheckMappings(Map<String, String> healthCheckMappings)
+	{
+		this.healthCheckMappings = healthCheckMappings;
+	}
+
```
Comment (ern) [inline providers/unboundid/src/java/org/sakaiproject/unboundid/UnboundidDirectoryProvider.java:1271]: please remove the useless javadoc

File: providers/component/src/webapp/WEB-INF/unboundid-ldap.xml (hunk 1 above)
Comment (austin48) [reply providers/component/src/webapp/WEB-INF/unboundid-ldap.xml]: done.  I rechecked the LDAPConnectionPoolHealthCheck class in UnboundId's 4.0.12 source code and and the default health mostly does

`// No processing is performed in this default implementation.`

and the only method that does something, ensureConnectionValidAfterException(), only throws an error if the connection is bad.

File: providers/unboundid/src/java/org/sakaiproject/unboundid/UnboundidDirectoryProvider.java (hunk 2 above)
Comment (austin48) [reply providers/unboundid/src/java/org/sakaiproject/unboundid/UnboundidDirectoryProvider.java]: done.  javadoc is removed

**Review by ottenhoff — APPROVED** [review]

**Pull request merged**